packages/indexer/src/indexers/index.ts

```diff
@@ -1,5 +1,95 @@
-
+import { eq, schema, usingDb } from "@metadaoproject/indexer-db";
+import {
+  IndexerImplementation,
+  IndexerType,
+} from "@metadaoproject/indexer-db/lib/schema";
+import { AccountInfoIndexer } from "./account-info-indexer";
+import { AmmMarketAccountUpdateIndexer } from "./amm-market/amm-market-account-indexer";
+import { connection } from "../connection";
+import { PublicKey } from "@solana/web3.js";
 
 export async function startIndexers() {
-  
-}
\ No newline at end of file
+  await startAccountInfoIndexers();
+  console.log("indexers successfully started");
+}
+
+export async function startAccountInfoIndexers() {
+  const accountInfoIndexers = await usingDb((db) =>
+    db
+      .select()
+      .from(schema.indexers)
+      .where(eq(schema.indexers.type, IndexerType.AccountInfo))
+      .fullJoin(
+        schema.indexerAccountDependencies,
+        eq(schema.indexerAccountDependencies.name, schema.indexers.name)
+      )
+      .execute()
+  );
+
+  for (const indexerQueryRes of accountInfoIndexers) {
+    startAccountInfoIndexer(indexerQueryRes);
+  }
+}
+
+type IndexerWithAccountDeps = {
+  indexers: {
+    name: string;
+    implementation: IndexerImplementation;
+    latestSlotProcessed: bigint;
+    type: IndexerType;
+  } | null;
+  indexer_account_dependencies: {
+    name: string;
+    acct: string;
+    latestTxSigProcessed: string | null;
+  } | null;
+};
+
+async function startAccountInfoIndexer(
+  indexerQueryRes: IndexerWithAccountDeps
+) {
+  const { indexers: indexer, indexer_account_dependencies: dependentAccount } =
+    indexerQueryRes;
+  if (!indexer) return;
+  const implementation = getIndexerImplementation(indexer.implementation);
+  if (implementation && dependentAccount && dependentAccount.acct) {
+    const accountPubKey = new PublicKey(dependentAccount.acct);
+
+    const accountInfo = await connection.getAccountInfo(accountPubKey);
+
+    //index refresh on startup
+    if (accountInfo) {
+      const res = await implementation.index(accountInfo, accountPubKey);
+      if (!res.success) {
+        console.error(
+          "error indexing account update",
+          accountPubKey.toString()
+        );
+      }
+    }
+
+    connection.onAccountChange(accountPubKey, async (accountInfo, context) => {
+      const res = await implementation.index(
+        accountInfo,
+        accountPubKey,
+        context
```

Comment:
here's where the parsing comes in?

Reply:
mhm if you peek inside that index() function for the specific implementation